packages/gatsby-remark-images/src/index.js

```diff
@@ -335,6 +335,12 @@ module.exports = (
   <figure class="gatsby-resp-image-figure" style="${wrapperStyle}">
     ${rawHTML}
     <figcaption class="gatsby-resp-image-figcaption">${imageCaption}</figcaption>
+  </figure>
```

Comment:
Can we try to reduce duplication of markup?

How about something like this?

```suggestion
  if (imageCaption || !options.wrapFigureOnlyWithCaptions) {
    if (imageCaption) {
      rawHTML = `
        ${rawHTML}
        <figcaption class="gatsby-resp-image-figcaption">${imageCaption}</figcaption>
      `
    }
    rawHTML = `
      <figure class="gatsby-resp-image-figure" style="${wrapperStyle}">
        ${rawHTML}
      </figure>
    `
  }
```

Alternatively one back to back, less indentation.

Reply:
I went with the if-else to try to make it more readable. I can change it as you suggested if you prefer so.